### ISSUE TITLE
Write tests

### ISSUE THREAD

**Issue by mProjectsCode** [mProjectsCode/obsidian-media-db-plugin#11]
- reintroduce the test api
- write some automated tests

**Comment by AB1908** [comment]
Hey, if you give me some guidelines, I'd be happy to give it a stab. I've contributed absolutely nothing to this plugin so this would be a good start.

**Comment by mProjectsCode** [comment]
@AB1908 do you want to hop on a discord call?

I have some rough ideas for tests, but I think I can explain them better on a discord call than here on GitHub

**Comment by AB1908** [comment]
Sure, what time is comfortable for you?

**Comment by AB1908** [comment]
I've been chipping away at this slower than I had hoped to but it sure is tricky to mock APIs! Would you have any examples? It's also strangely unsatisfying to check logic and not add any features haha but I suppose I'll get a chance to do that once some sanity tests are in place.

**Comment by mProjectsCode** [comment]
I don't have any examples for you for mocking/testing APIs, but you can look at these tests that I wrote. https://github.com/mProjectsCode/obsidian-meta-bind-plugin/tree/master/tests

I can also create a test branch and you could open a PR to it so I can help you with that.

**Comment by AB1908** [comment]
I'll clean up and open a draft PR then.

**Comment by AB1908** [comment]
Wohoo! First round complete.

I'll look at where else testing would be useful and then maybe I'll finally start refactoring but meanwhile do you have any parts of the codebase in mind?

**Comment by mProjectsCode** [comment]
I am currently doing some heavy refactoring of `main.ts` and the modals myself.
Maybe we should jump on a discord call to discuss ideas and things that need to be done.
Then I can also share my plans for this plugin during obsidian October with you.

**Comment by AB1908** [comment]
That sounds fantastic. Hit me up on Discord and we can arrange something.

**Comment by mProjectsCode** [comment]
I just saw that you added credits to fevol for the fetch mock in the commit message. We should also add it to the top of the file and the credits section in the readme.

**Comment by AB1908** [comment]
Agreed! Didn't strike me at the time but I'll make sure I get to that.
